lib/forwardanalyzer.cpp

```diff
@@ -529,11 +529,11 @@ struct ForwardTraversal {
         return updateLoop(endToken, endBlock, condTok, initTok, stepTok, true);
     }
 
-    Progress updateScope(Token* endBlock) {
-        return updateRange(endBlock->link(), endBlock);
+    Progress updateScope(Token* endBlock, int depth = 5) {
+        return updateRange(endBlock->link(), endBlock, depth);
     }
 
-    Progress updateRange(Token* start, const Token* end, int depth = 20) {
+    Progress updateRange(Token* start, const Token* end, int depth = 5) {
```

Comment:
5 seems too small. There is a lot of code that uses nesting deeper than 5.

Reply:
What would a test case look like that would fail with `depth` = 5?